### PR TITLE
Databases week03 homework

### PR DESCRIPTION
**Pull request by Laurelinduilwen**
Explanation is in readme-todoapp.md :(

### CODE REVIEW

File: Week3/sqlqueries.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const database =
+  'CREATE DATABASE IF NOT EXISTS todoapp DEFAULT CHARACTER SET utf8 COLLATE utf8_general_ci';
+const useDatabase = 'use todoapp';
+
+const users =
+  'create table IF NOT EXISTS users (id int PRIMARY KEY, username varchar(50), password varchar(50), email varchar(50), created_at timestamp)';
+const todolists =
+  'create table IF NOT EXISTS todolists (id int PRIMARY KEY, list_name varchar(255), created_at datetime DEFAULT now(), user_id int)';
+const tasks =
+  "create table IF NOT EXISTS tasks (id int PRIMARY KEY, name varchar(255), task_id int NOT NULL, status ENUM('completed', 'not_completed'), created_at datetime DEFAULT now())";
```
Comment (mogaleaf) [inline Week3/sqlqueries.js]: Task_id should be better as List_id :)

File: Week3/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+app.post('/auth', function(request, response) {
+  const username = request.body.username;
+  const password = request.body.password;
+  if (username && password) {
+    connection.query(
+      'SELECT * FROM users WHERE username = ? AND password = ?',
+      [username, password],
+      function(error, results, fields) {
+        if (results.length > 0) {
```
Comment (mogaleaf) [inline Week3/index.js]: Check the error first here, if an error is raised result will be undefined.

**Comment by Laurelinduilwen** [comment]
Sorry for commenting this late, this homework kinda threw off my whole trajectory :) 

I think most of the problems I am having with it are related to express and my lack of experience in dealing with the errors that node is throwing at me. It will take a bit for me to fix this mess...

File: Week3/index.js (hunk 2 above)
Comment (Laurelinduilwen) [reply Week3/index.js]: I actually had it right there but removed it to simplify the code a bit, will add it back again :)

**Pull request closed without merging**